digdag-core/src/main/java/io/digdag/core/agent/RequireOperatorFactory.java

```diff
@@ -65,43 +68,149 @@ private RequireOperator(OperatorContext context, TaskCallbackApi callback)
+                if (!requireKicked &&  (
+                        rerunOn == OptionRerunOn.ALL ||
+                                rerunOn == OptionRerunOn.FAILED && !attempt.getStateFlags().isSuccess())) {
+
+                    //To force run, set flag gen_retry_attempt_name and do polling
+                    throw nextPolling(lastStateParams.deepCopy().set("rerun_on_retry_attempt_name", UUID.randomUUID().toString()));
+                }
+                if (!ignoreFailure && !attempt.getStateFlags().isSuccess()) {
+                    // ignore_failure is false and the attempt is in error state. Make this operator failed.
+                    throw new TaskExecutionException(String.format(ENGLISH,
+                            "Dependent workflow failed. Session id: %d, attempt id: %d",
+                            attempt.getSessionId(), attempt.getId()));
+                }
+                return TaskResult.empty(cf);
+            }
+            else {
+                // Wait for finish running attempt
+                throw nextPolling(request.getLastStateParams().deepCopy());
+            }
+        }
+
+        private TaskExecutionException nextPolling(Config stateParams)
+        {
+            int iteration = stateParams.get("retry", int.class, 0);
+            stateParams.set("retry", iteration+1);
+            int interval = (int) Math.min(1 * Math.pow(2, iteration), MAX_TASK_RETRY_INTERVAL);
+
+            return TaskExecutionException.ofNextPolling( interval, ConfigElement.copyOf(stateParams));
+        }
+
+        /**
+         * Make ProjectIdentifier from parameters.
+         * @return
+         */
+        private ProjectIdentifier makeProjectIdentifier()
+        {
+            Config config = request.getConfig();
+            Config localConfig = request.getLocalConfig();
+            int projectId = config.get("project_id", int.class);
+            Optional<Integer> projectIdParam = localConfig.getOptional("project_id", Integer.class);
+            Optional<String> projectNameParam = localConfig.getOptional("project_name", String.class);
+
+            if (projectIdParam.isPresent() && projectNameParam.isPresent()) {
+                throw new ConfigException("Both project_id and project_name can't be set");
+            }
+            ProjectIdentifier projectIdentifier;
+            if (projectNameParam.isPresent()) {
+                projectIdentifier = ProjectIdentifier.ofName(projectNameParam.get());
+            }
+            else if (projectIdParam.isPresent()) {
+                projectIdentifier = ProjectIdentifier.ofId(projectIdParam.get());
+            }
+            else {
+                projectIdentifier = ProjectIdentifier.ofId(projectId);
+            }
+            return projectIdentifier;
+        }
+    }
+
+    enum OptionRerunOn
+    {
+        NONE("none"),
+        FAILED( "failed"),
+        ALL ("all");
+
+        private final String name;
```

Comment:
This isn't needed?

Reply:
Ah, correct. Currently not used. Will remove it. Thanks!